addons/website/models/website.py

```diff
@@ -447,6 +447,17 @@ def get_url_localized(router, lang):
     # Utilities
     # ----------------------------------------------------------
 
+    @api.model
+    def _get_request_website(self):
+        """ Return the website set on `request` if called in a frontend context
+        (website=True on route).
+        This method can typically be used to check if we are in the frontend.
+
+        This method is easy to mock during python tests to simulate frontend
+        context, rather than mocking every method accessing request.website.
+        """
+        return request and hasattr(request, 'website') and request.website or False
```

Comment:
Done.
It comes with a cost, as now it is not a Class method anymore, we need to import it to use it, but we can't import it directly or the mock won't work as import in files will 'override' it:
It became
```python
from odoo.addons.website.models import ir_http
ir_http.get_request_website()
```
Instead of
```python
self.env['website'].get_request_website()
```
See [this blog](https://blog.swdev.ed.ac.uk/2017/04/06/how-you-import-impacts-how-you-mock/). 
I am ok with that, let me know if something bugs you